app/models/company.rb

```diff
@@ -2,4 +2,32 @@ class Company < ActiveRecord::Base
   validates :name, :city, presence: true
   validates :name, uniqueness: true
   has_many :jobs
+  has_many :contacts
+
+  #sorts all companies by location
+  def self.sort_by_city
+    order(:city)
+  end
+
+  def self.top_three_companies_by_interest
+    sorted = []
+    all.each do |company|
+      sorted << [company.jobs.average(:level_of_interest), company]
+    end
+    sorted.sort_by{|interest, company| interest}[0..2].reverse
```

Comment:
there's a lot of stuff going on in this method - can we break it into a few smaller methods?